src/System.Windows.Forms/src/System/Windows/Forms/ToolTip.cs

```diff
@@ -687,6 +687,27 @@ private void CheckNativeToolTip(Control associatedControl)
             }
         }
 
+        private void CheckNativeCompositeControls(Control associatedControl, string text)
+        {
+            if (!(associatedControl is TabPage))
```

Comment:
Could use new language features to avoid the secondary as-cast below by naming the result immediately
```suggestion
            if (!(associatedControl is TabPage tabPage))
```

Reply:
I thought It won't work. Thank you! Fixed.